crates/exex/src/atlantic/endpoints.rs

```diff
@@ -0,0 +1,266 @@
+
+        // Check that URLs are constructed correctly even with a trailing slash in base URL
+        assert_eq!(
+            endpoints.proof_generation,
+            Url::parse("https://example.com/proof-generation").unwrap()
+        );
+        assert_eq!(
+            endpoints.trace_generation,
+            Url::parse("https://example.com/trace-generation").unwrap()
+        );
+        assert_eq!(
+            endpoints.trace_generation_proof_generation,
+            Url::parse("https://example.com/trace-generation-proof-generation").unwrap()
+        );
+        assert_eq!(
+            endpoints.atlantic_query,
+            Url::parse("https://example.com/l2/atlantic-query").unwrap()
+        );
+        assert_eq!(
+            endpoints.proof_generation_verification,
+            Url::parse("https://example.com/l2/atlantic-query/proof-generation-verification")
+                .unwrap()
+        );
+        assert_eq!(
+            endpoints.proof_verification,
+            Url::parse("https://example.com/l2/atlantic-query/proof-verification").unwrap()
+        );
+    }
+
+    #[test]
+    fn test_l2_endpoints_valid_base_url() {
+        let base_url = Url::parse("https://example.com").expect("Failed to parse base URL");
+        let endpoints = L2Endpoints::new(&base_url).expect("Failed to create L2Endpoints");
+
+        // Check that URLs are constructed correctly even without a trailing slash in base URL
+        assert_eq!(
+            endpoints.proof_generation,
+            Url::parse("https://example.com/proof-generation").unwrap()
+        );
+        assert_eq!(
+            endpoints.trace_generation,
+            Url::parse("https://example.com/trace-generation").unwrap()
+        );
+        assert_eq!(
+            endpoints.trace_generation_proof_generation,
+            Url::parse("https://example.com/trace-generation-proof-generation").unwrap()
+        );
+        assert_eq!(
+            endpoints.atlantic_query,
+            Url::parse("https://example.com/l2/atlantic-query").unwrap()
+        );
+        assert_eq!(
+            endpoints.proof_generation_verification,
+            Url::parse("https://example.com/l2/atlantic-query/proof-generation-verification")
+                .unwrap()
+        );
+        assert_eq!(
+            endpoints.proof_verification,
+            Url::parse("https://example.com/l2/atlantic-query/proof-verification").unwrap()
+        );
+    }
+
+    #[test]
+    fn test_sharp_queries_endpoints_invalid_base_url() {
+        // Test with an invalid base URL
+        let base_url = Url::parse("invalid-url");
+        assert!(base_url.is_err(), "Expected base URL parsing to fail");
+    }
+
+    #[test]
+    fn test_sharp_queries_endpoints_valid_base_url_trailing_slash() {
+        let base_url = Url::parse("https://example.com/")
+            .expect("Failed to parse base URL with trailing slash");
```

Comment:
see above

Reply:
same, this is unwrap with a nice message